log-warper/src/Log/Name.hs

```diff
@@ -0,0 +1,59 @@
+{-# LANGUAGE UndecidableInstances #-}
+
+-- | Contains @newtype@ wrapper 'LoggerName' for list of components for logger name.
+
+module Log.Name
+       ( LoggerName (..)
+       ) where
+
+import Universum
+
+import Data.Aeson.Types (ToJSON, ToJSONKey (..))
+import Fmt (build, fmt)
+import GHC.Exts (IsList (..))
+import Text.Show (Show (show))
+
+import qualified Data.Text as Text
+
+{- | Logger name is a unique label for some 'LogAction'. Loggers in @log-warper@
+form tree. Every 'LoggerName' contains some list of components. In textual
+representation those commponents separated by dot character. Below you can see
+example of correspondence between list of components and string:
+
+@
+["foor", "bar", "baz"] <-> "foo.bar.baz"
+@
+-}
+newtype LoggerName = LoggerName
+    { unLoggerName :: Vector Text
+    } deriving (Eq, Ord, Generic, Semigroup, Monoid, Container)
```

Comment:
`Vector`, why not list? (Just from performance perspective -- I noted IsList below)